R/article.R

```diff
@@ -281,19 +281,67 @@ mnras_article <- function(..., keep_tex = TRUE, fig_caption = TRUE) {
   )
 }
 
+#' @param document_style For \code{oup_article}, one of "contemporary", "modern", or "traditional" setting overall style of document. Defaults to "contemporary".
+#' @param papersize For \code{oup_article}, one of "large", "medium", or "small" setting output page size. Defaults to "large".
+#' @param namedate For \code{oup_article}, a logical variable indicating if natbib citations should be in name-date format. Defaults to \code{FALSE}.
+#' @param onecolumn For \code{oup_article}, a logical variable indicating if one column formatting should be used. Defaults to \code{FALSE}.
 #' @section \code{oup_article}: Format for creating submissions to many Oxford University Press
 #'   journals. Adapted from
 #'   \url{https://academic.oup.com/journals/pages/authors/preparing_your_manuscript}
-#'   and \url{https://academic.oup.com/icesjms/pages/General_Instructions}.
+#'   and the \code{oup-authoring-template} available on CTAN at
+#'   \url{https://www.ctan.org/pkg/oup-authoring-template}.
 #' @export
 #' @rdname article
 oup_article <- function(
   ..., keep_tex = TRUE,
-  md_extensions = c("-autolink_bare_uris")
+  md_extensions = c("-autolink_bare_uris"),
+  journal=NULL,pandoc_args=NULL,
+  number_sections=TRUE,
+  citation_package=c("natbib","default"),
+  papersize=c("large","medium","small"),
+  document_style=c("contemporary","modern","traditional"),
+  namedate=FALSE,onecolumn=FALSE
 ) {
+  citation_package <- match.arg(citation_package)
+  if (length(citation_package) > 1) stop("Please choose just one ", dQuote("citation_package"))
```

Comment:
OK, will change.

Reply:
Note that I got this code from a line in the `ims_article` format. This would presumably also need to be corrected, but I am not going to do it as I don't want to unnecessarily mess around in someone else's code.

Reply:
It is now done f171c8f0

thanks for the pointer.